### DISCUSSION TITLE
Fixed Cost Inputs

### DISCUSSION THREAD

**Discussion by adam-morse** [NREL/REopt-API-Analysis#146] [5 - Financial Input Questions] [answered]
Hi NREL,

I haven’t seen a means to incorporate fixed or lump sum costs into the ReOpt model which I’ve relied on using other tools. For instance costs like microgrid controllers or utility studies or upgrades. These are  hard to conceptualize in terms of technology $/W. I use the fixed costs to help guide the optimization toward the efficiencies of scale benefit of larger systems. 

I think I saw some mention in the manual about modeling fixed, non technology specific costs, for offgrid systems. Is there any way to do this?

Adam Morse

**Comment by adfarth** [comment, marked as answer]
You can utilize the following inputs to capture some of these costs: 

- In the codebase only, and for grid-connected **resilience** runs: `Financial.microgrid_upgrade_cost_fraction`. REopt compares the expected outage costs to this microgrid upgrade cost (which is input as a % of total technology CAPEX) to determine if it is cost-optimal to incur any additional costs needed to island the system during a grid outage. 
- In the webtool and codebase, and only for **off-grid runs** (when "Grid" is unchecked): 
    - Webtool inputs: Additional capital costs and Additional annual costs under the "Financial" accordion
    - Codebase inputs: `Financial.offgrid_other_capital_costs` and `Financial.offgrid_other_annual_costs`

We don't currently have similar options for web tool resilience or cost savings runs. Your feedback is helpful for encouraging us to create those!
